akka-stream/src/test/scala/neotypes/akkastreams/AkkaStreamSpec.scala

```diff
@@ -2,25 +2,31 @@ package neotypes.akkastreams
 
 import akka.actor.ActorSystem
 import akka.stream.scaladsl.Sink
-import neotypes.BaseIntegrationSpec
+import neotypes.{Async, StreamIntegrationSpec}
 import neotypes.akkastreams.implicits._
-import neotypes.implicits.mappers.results._
-import neotypes.implicits.syntax.string._
-
 import scala.concurrent.Future
+import scala.concurrent.duration.Duration
+
+class AkkaStreamSpec extends StreamIntegrationSpec[AkkaStream, Future] {
+  implicit val system =
+    ActorSystem("QuickStart")
 
-class AkkaStreamSpec extends BaseIntegrationSpec[Future] {
-  ignore should "work with Akka streams" in execute { s =>
-    implicit val system = ActorSystem("QuickStart")
+  override def fToFuture[T](future: Future[T]): Future[T] =
+    future
 
-    "match (p:Person) return p.name"
-      .query[Int]
-      .stream[AkkaStream](s)
-      .runWith(Sink.seq[Int])
-      .map {
-        names => assert(names == (0 to 10))
+  override def streamToFList[T](stream: AkkaStream[T]): Future[List[T]] =
+    stream.runWith(Sink.seq[T]).map(_.toList).flatMap { result =>
+      Future {
+        // Delaying the answer a little seems to fix:
+        // https://github.com/neotypes/neotypes/issues/47
+        Thread.sleep(1000)
+        result
```

Comment:
Sometimes it's unavoidable, but Thread.sleep can cause issues down the road.  It's good to know that the issue is a timing thing though.

Reply:
Futures are hard to reason about. I still do not understand what exactly is the problem, but it seems that if this operation is too fast, it tries to close the transaction twice, which results in an error.

A small delay of 1 second is enough to fix the problem.
In general, I would not be happy with this kind of solution, but since this is just a unique test case, I would just leave it.